c-version/elrond/context.h

```diff
@@ -0,0 +1,59 @@
+#ifndef _CONTEXT_H_
+#define _CONTEXT_H_
+
+#include "types.h"
+
+void getSCAddress(byte *address);
+void getOwnerAddress(byte *address);
+int getShardOfAddress(const byte *address);
+int isSmartContract(const byte *address);
+
+// Call-related functions
+void getOriginalTxHash(byte *txHash);
+void getCaller(byte *callerAddress);
+int getFunction(byte *function);
+int getCallValue(byte *result);
+long long getGasLeft();
+void finish(const byte *data, int length);
+void int64finish(long long value);
+void writeLog(byte *pointer, int length, byte *topicPtr, int numTopics);
+void asyncCall(const byte *destination, const byte *value, const byte *data, int length);
+void signalError(byte *message, int length);
+
+int executeOnSameContext(long long gas, byte *address, byte *value, byte *function, int functionLength, int numArguments, byte *argumentsLengths, byte *arguments);
+int executeOnDestContext(long long gas, byte *address, byte *value, byte *function, int functionLength, int numArguments, byte *argumentsLengths, byte *arguments);
+int createContract(long long gas, byte *value, byte *code, int codeSize, byte *newAddress, int numInitArgs, byte *initArgLengths, byte *initArgs);
+
+// Blockchain-related functions
+long long getBlockTimestamp();
+long long getBlockNonce();
+int getBlockHash(long long nonce, byte *hash);
+
+// Argument-related functions
+int getNumArguments();
+int getArgument(int argumentIndex, byte *argument);
+long long int64getArgument(int argumentIndex);
+int getArgumentLength(int argumentIndex);
+
+// Account-related functions
+void getExternalBalance(const byte *address, byte *balance);
+int transferValue(const byte *destination, const byte *value, const byte *data, int length);
+
+// Storage-related functions
+int storageLoadLength(const byte *key, int keyLength);
+int storageStore(const byte *key, int keyLength, const byte *data, int dataLength);
+int storageLoad(const byte *key, int keyLength, byte *data);
+int int64storageStore(const byte *key, int keyLength, long long value);
+long long int64storageLoad(const byte *key, int keyLength);
```

Comment:
int64storageStore and int64storageLoad are deprecated, we should remove them from here from all contracts

I'm posting here the small int EI from arwen:

// extern long long smallIntGetUnsignedArgument(void *context, int32_t id);
// extern long long smallIntGetSignedArgument(void *context, int32_t id);
//
// extern void smallIntFinishUnsigned(void* context, long long value);
// extern void smallIntFinishSigned(void* context, long long value);
//
// extern int32_t smallIntStorageStoreUnsigned(void *context, int32_t keyOffset, int32_t keyLength, long long value);
// extern int32_t smallIntStorageStoreSigned(void *context, int32_t keyOffset, int32_t keyLength, long long value);
// extern long long smallIntStorageLoadUnsigned(void *context, int32_t keyOffset, int32_t keyLength);
// extern long long smallIntStorageLoadSigned(void *context, int32_t keyOffset, int32_t keyLength);
//
// ... and 4 deprecated functions that we should no longer have in the headers